react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      count: 0
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({
+      count: this.state.count + 1
+    });
+  }
+
+  render() {
+    let button;
+    const currentCount = this.state.count;
+    if (currentCount <= 2) {
+      button = <button className="button" onClick={this.handleClick}>Hot Button</button>;
+    } else if (currentCount >= 3 && currentCount <= 5) {
+      button = <button className="button2" onClick={this.handleClick}>Hot Button</button>;
+    } else if (currentCount >= 6 && currentCount <= 8) {
+      button = <button className="button3" onClick={this.handleClick}>Hot Button</button>;
+    } else if (currentCount >= 9 && currentCount <= 11) {
+      button = <button className="button4" onClick={this.handleClick}>Hot Button</button>;
+    } else if (currentCount >= 12 && currentCount <= 14) {
+      button = <button className="button5" onClick={this.handleClick}>Hot Button</button>;
+    } else if (currentCount >= 15) {
+      button = <button className="button6" onClick={this.handleClick}>Hot Button</button>;
+    }
+    return (
+      button
+    );
```

Comment:
Although this works and is a perfectly acceptable answer, another way to think about it to save some typing and make it easier to maintain in the future would be to just use the if statement to choose the class name. Every part of the button is the exact same except for the class name. So for example you could do something like:

```jsx
let btnClass;
const currentCount = this.state.count;
if (currentCount <= 2) {
  btnClass = 'button';
} else if (currentCount >= 3 && currentCount <= 5) {
  btnClass = 'button2';
} else if (currentCount >= 6 && currentCount <= 8) {
  btnClass = 'button3';
} else if (currentCount >= 9 && currentCount <= 11) {
  btnClass = 'button4';
} else if (currentCount >= 12 && currentCount <= 14) {
  btnClass = 'button5';
} else if (currentCount >= 15) {
  btnClass = 'button6';
} 

return (
  <button className={btnClass} onClick={this.handleClick}>Hot Button</button>
);

Reply:
Thanks Scott,

Knew I was making things a little too busy in those if statements, thanks for the advice on how to clean it up!